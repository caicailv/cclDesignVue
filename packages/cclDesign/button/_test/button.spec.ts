import { mount } from '@vue/test-utils'
import { describe, expect, test } from 'vitest'
import { CButton } from '../index'

describe('CButton', () => {
  test('class', async () => {
    const wrapper = mount(CButton, {
      props: {
        type: 'secondary',
      },
    })
    expect(wrapper.classes()).toContain('c-button')
    expect(wrapper.classes()).toContain('c-button--secondary')
    await wrapper.setProps({ status: 'success' })
    expect(wrapper.classes()).toContain('c-button--success')
  })
  test('click', async () => {
    const wrapper = mount(CButton, {
      props: {
        disabled: true,
      },
    })
    wrapper.find('button').trigger('click')
    expect(wrapper.emitted('click')).toBeUndefined()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
