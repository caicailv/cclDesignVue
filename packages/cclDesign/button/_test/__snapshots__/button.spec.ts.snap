// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CButton > click 1`] = `"<button disabled=\\"\\" class=\\"c-button c-button--primary c-button--disabled\\"></button>"`;

exports[`CButton 1`] = `"<button class=\\"c-button c-button--secondary\\"></button>"`;
